code/modules/mining/mine_items.dm

```diff
@@ -332,13 +332,45 @@
 	anchored = 1
 	density = 1
 	var/arbitraryatmosblockingvar = TRUE
+	var/buildstacktype = /obj/item/stack/sheet/metal
+	var/buildstackamount = 5
+
+/obj/structure/fans/deconstruct()
+	if(buildstacktype)
+		new buildstacktype(loc,buildstackamount)
+	..()
+
+/obj/structure/fans/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/wrench) && !(flags&NODECONSTRUCT))
+		playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
+		user.visible_message("<span class='warning'>[user] disassembles the fan.</span>", \
+						"<span class='notice'>You start to disassemble the fan...</span>", "You hear clanking and banging noises.")
+		if(do_after(user, 20/W.toolspeed, target = src))
+			deconstruct()
+			return ..()
 
 /obj/structure/fans/tiny
 	name = "tiny fan"
 	desc = "A tiny fan, releasing a thin gust of air."
 	layer = TURF_LAYER + 0.8
 	density = 0
 	icon_state = "fan_tiny"
+	var/buildstacktype1 = /obj/item/stack/sheet/metal
+	var/buildstackamount1 = 2
+
+/obj/structure/fans/tiny/deconstruct()
+	if(buildstacktype1)
+		new buildstacktype1(loc,buildstackamount1)
+	..()
+
+/obj/structure/fans/tiny/attackby(obj/item/weapon/W, mob/user, params)
```

Comment:
This proc is identical to the parent /obj/structure/fans/attackby proc above, children inherit the procs of their parent, this is what those / in the type specifies, a hierarchy of proc and var inheritance.

Because of this, this whole proc can be removed.